pkg/asset/manifests/infrastructure.go

```diff
@@ -80,6 +80,27 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 		},
 	}
 
+	if installConfig.Config.ControlPlane.Replicas != nil && *installConfig.Config.ControlPlane.Replicas < 3 {
+		config.Status.ControlPlaneTopology = configv1.SingleReplicaTopologyMode
+	} else {
+		config.Status.ControlPlaneTopology = configv1.HighlyAvailableTopologyMode
+	}
+
+	numOfWorkers := int64(0)
+	for _, mp := range installConfig.Config.Compute {
+		if mp.Replicas != nil {
+			numOfWorkers += *mp.Replicas
+		}
+	}
+	switch numOfWorkers {
+	case 0:
+		config.Status.InfrastructureTopology = config.Status.ControlPlaneTopology
```

Comment:
Is this:

> We expect the "infrastructure" workload to be schedulable on the control-plane nodes in this compact cluster.

?  Because in that case it may be worth switching on `schedulers.config.openshift.io`'s `mastersSchedulable`.

Reply:
Although maybe in the `mastersSchedulable` case, operators that would usually schedule their operands should look at `mastersSchedulable`, realize they were being allowed onto the control-plane machines, and, based on [the API docs][1] consume `controlPlaneTopology` instead of `infrastructureTopology`?

[1]: https://github.com/openshift/api/pull/827/files#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR88-R9

Reply:
It's already happens. If you install only masters nodes, there are schedualable

Reply:
https://github.com/openshift/installer/blob/master/pkg/asset/manifests/scheduler.go#L64-L74

Reply:
Right.  My concern is that you are assuming that the `scheduler.go` code is going to keep doing what it's doing today.  I'd rather have this code look at `Scheduler.Spec.MastersSchedulable`, so it didn't have to make assumptions about `scheduler.go`.  And I'm still not sure if `infrastructureTopology` matters at all, or who it controls, if `mastersSchedulable` is `true`.

Reply:
That's not a bad suggestion. It would be nice to de-duplicate the logic. I think that I would rather have a common function that the two assets could each call to determine whether the masters should be considered schedulable as opposed to having the Infrastructure asset carry a dependency on the Scheduler asset, though.